calblueprint/ogsc
# User accept invite flow

# Creating the user accept invite flow
- Adds a migration to `userInvites` table
- New flow for user who has received an email to accept their invite to our platform
- Form will have info filled out by default & user can update/add in their name, phone number, and password (can't change email)
- Sendgrid accept request template has been updated with the correct URL

### 12/4 Note
- Resetting little state machine after pressing submit to avoid confusion for second time accepting an invite (for example: if my parents go through this flow on the same browser, the info from parent1 would autofill when parent2 is trying to accept the account)
- Did not save password in state (was having issues with saving password specifically in little state machine for some reason + would have saved just to reset it again). Please comment/lmk if anyone thinks we should still save password!

## TODO 

- [x] Error page if the email/invite has already been accepted
- [x] Error page if they have invalid invite code
- [x] Update `show/hide passwords` on page 2 to function separately
- [ ] After accepting invite, where should the user be directed to? (their dashboard doesn't exist yet)

## How to Test/Use PR feature
(locally/without actually sending an email)
- Send an invite at http://localhost:3000/admin/invite/new
- Find the invite code id at `id` in `userInvites`
- Go to http://localhost:3000/users/acceptInvite?inviteCode={YOUR_INVITE_CODE_ID}
- Check that your name is right in the title + Enter/verify your information
- Check that the `acceptedAt` field in `userInvites` is no longer null/`hashedPassword` in `users` has a value

## PR Checklist

Does your branch (check if true):


## Migrations

- Added an `acceptedAt` field to `userInvites` table which updates once the user has gone through this flow and accepted the invite
- Run `yarn db:migrate up`
 
## Screenshots

<img width="1417" alt="Screen Shot 2020-11-28 at 4 31 41 PM" src="https://user-images.githubusercontent.com/34170094/100529178-7f464100-3199-11eb-912d-23329e511895.png">

- Correct URL
- Correct name in the title
- Info is pre-filled out

<img width="1418" alt="Screen Shot 2020-11-28 at 4 31 51 PM" src="https://user-images.githubusercontent.com/34170094/100529194-a735a480-3199-11eb-972f-7bf610b93e9c.png">

### Error Pages
No Access: invalid invite code or API fails
![Screen Shot 2020-12-04 at 10 34 59 PM](https://user-images.githubusercontent.com/34170094/101236063-cdb87b80-3682-11eb-8c24-8107d0097590.png)
Expired: invite code already accepted or invite expired (don't think this is implemented yet) 
![Screen Shot 2020-12-04 at 10 35 11 PM](https://user-images.githubusercontent.com/34170094/101236065-d01ad580-3682-11eb-8004-8fe145a62851.png)



CC: @erinysong
